lib/diplomat/rest_client.rb

```diff
@@ -304,6 +306,14 @@ def send_delete_request(connection, url, options, custom_params = nil)
       end
     end
 
+    # This method returns the correct Exception Classes depending on the Faraday version being installed
+    # see https://github.com/WeAreFarmGeek/diplomat/issues/227
+    #
+    # @return [Array<Class>] Faraday error classes that should be caught
+    def faraday_error_classes
+      Faraday.const_defined?(:ServerError) ? [Faraday::ClientError, Faraday::ServerError] : [Faraday::ClientError]
```

Comment:
Hum, we probably misunderstood each other:

My point was:
 * Either we just want to catch all, in such case, we just catch Faraday::Error and this method is not useful => which is exactly what you do (because in Faraday 1+: Faraday::Error := Faraday::ClientError + Faraday::ServerError while in 0.9, Faraday::Error := Faraday::ClientError which includes also all errors of Faraday::ServerError )
 * either you create this method, but in such case, it should *only* return Faraday::ServerError when defined and Faraday::ClientError on Faraday 0.9